features/storage/TESTS/kvstore/direct_access_devicekey_test/main.cpp

```diff
@@ -268,15 +268,12 @@ void test_direct_access_to_device_inject_root()
         internal_start_address =  MBED_CONF_STORAGE_TDB_INTERNAL_INTERNAL_BASE_ADDRESS;
         internal_rbp_size =  MBED_CONF_STORAGE_TDB_INTERNAL_INTERNAL_SIZE;
     } else if (strcmp(STR(MBED_CONF_STORAGE_STORAGE_TYPE), "default") == 0) {
-#if COMPONENT_QSPIF || COMPONENT_SPIF || COMPONENT_DATAFLASH
-        internal_start_address =  MBED_CONF_STORAGE_TDB_EXTERNAL_INTERNAL_BASE_ADDRESS;
-        internal_rbp_size =  MBED_CONF_STORAGE_TDB_EXTERNAL_RBP_INTERNAL_SIZE;
+#if COMPONENT_FLASHIAP
+        internal_start_address =  MBED_CONF_STORAGE_TDB_INTERNAL_INTERNAL_BASE_ADDRESS;
+        internal_rbp_size =  MBED_CONF_STORAGE_TDB_INTERNAL_INTERNAL_SIZE;
 #elif COMPONENT_SD
```

Comment:
COMPONENT_SD is also not needed and a compilation error should be emitted if COMPONENT_FLASHIAP is not defined.